cql3/restrictions/partition_key_restrictions.cc

```diff
@@ -52,5 +53,70 @@ const expr::single_column_restrictions_map& partition_key_restrictions::get_sing
 bool partition_key_restrictions::partition_key_restrictions_is_empty() const {
     return is_empty_restriction(_partition_restrictions);
 }
+
+static bool contains_column(const expression& e) {
+    return find_in_expression<column_value>(e, [](const column_value&) { return true; }) != nullptr;
+}
+
+void partition_key_restrictions::analyze_restrictions() {
```

Comment:
I think it should be in a more general form, a function that gets an expression and returns a struct. This will make it easier to write unit tests.

Reply:
Hmm I didn't take unit tests into account, I will try to change it to be easier to test.